### DISCUSSION TITLE
Can't get solid to run with OutputDefinition

### DISCUSSION THREAD

**Discussion by stejul** [dagster-io/dagster#4732] [Q&A] [answered]
Hello!

I found Dagster recently and wanted to try it out by creating a small example project. While creating the solid and configuring the OutputDefinition. I always get this error when i fill out the `output_defs` parameters in my solid. Should I just skip on using `output_defs`?
![image](https://user-images.githubusercontent.com/7622701/132032143-93f755b9-07c6-4d45-aa23-f8e53cddb7ff.png)

This is my code example, that I can't get to run:
```python
from dagster import Output, InputDefinition, OutputDefinition, solid, pipeline, ModeDefinition, List as DagsterList, String
from os import walk

@solid(name="getListOfFiles", output_defs=OutputDefinition(dagster_type=DagsterList[String]))
def get_all_csv_files(context) -> DagsterList[String]:
    result: DagsterList[String] = []
    for (dirpath, dirname, filenames) in walk("src/data/"):
        for file in filenames:
            context.log.info(f"Found following file in directory: src/data/{file}")
            result.append(f"src/data/{file}")
    return result

@pipeline()
def execute_pipeline():
    get_all_csv_files()

```
Help would be very much appreciated 😄

**Comment by sryza** [comment, marked as answer]
Hi @stejul - if you replace `OutputDefinition(dagster_type=DagsterList[String])` with `[OutputDefinition(dagster_type=DagsterList[String])]` (i.e. surround it in a list), it should work.

**Reply by stejul** [reply]
Hi @sryza, yup it works now. Thank you so much!
